site/app/controllers/forum/ForumController.php

```diff
@@ -1185,128 +1136,77 @@ public function getSingleThread() {
-                    // Redirect merged thread to parent
-                    $this->core->addSuccessMessage("Requested thread is merged into current thread.");
-                    if (!empty($_REQUEST["ajax"])) {
-                        return $this->core->getOutput()->renderJsonSuccess(['merged' => true, 'destination' => $this->core->buildCourseUrl(['forum', 'threads', $thread['merged_thread_id']])]);
-                    }
-                    $this->core->redirect($this->core->buildCourseUrl(['forum', 'threads', $thread['merged_thread_id']]));
-                    return;
-                }
-                if ($option == "alpha") {
-                    $posts = $this->core->getQueries()->getPostsForThread($current_user, $thread_id, $show_deleted, 'alpha');
-                }
-                elseif ($option == "alpha_by_registration") {
-                    $posts = $this->core->getQueries()->getPostsForThread($current_user, $thread_id, $show_deleted, 'alpha_by_registration');
-                }
-                elseif ($option == "alpha_by_rotating") {
-                    $posts = $this->core->getQueries()->getPostsForThread($current_user, $thread_id, $show_deleted, 'alpha_by_rotating');
-                }
-                elseif ($option == "reverse-time") {
-                    $posts = $this->core->getQueries()->getPostsForThread($current_user, $thread_id, $show_deleted, 'reverse-time');
-                }
-                else {
-                    $posts = $this->core->getQueries()->getPostsForThread($current_user, $thread_id, $show_deleted, 'tree');
-                }
-                if (empty($posts)) {
-                    $this->core->addErrorMessage("No posts found for selected thread.");
-                }
-            }
-        }
-        if (empty($thread_id) || empty($posts)) {
-            $new_posts = $this->core->getQueries()->getUnviewedPosts(-1, $current_user);
-            $posts = $this->core->getQueries()->getPostsForThread($current_user, -1, $show_deleted);
-        }
-        $thread_id = -1;
-        if (!empty($posts)) {
-            $thread_id = $posts[0]["thread_id"];
+
+        $repo = $this->core->getCourseEntityManager()->getRepository(Thread::class);
+        $thread = $repo->getThreadDetail($thread_id, $option, $show_deleted);
+        if (is_null($thread)) {
+            return $this->core->getOutput()->renderJsonFail("Invalid thread id (NON-EXISTENT ID)");
         }
-        foreach ($posts as &$post) {
-            do {
-                $post['content'] = preg_replace('/(?:!\[(.*?)\]\((.*?)\))/', '$2', $post['content'], -1, $count);
-            } while ($count > 0);
+
+        $this->core->getQueries()->markNotificationAsSeen($user, -2, (string) $thread_id);
+        if ($thread->isMergedThread()) {
+            // Redirect merged thread to parent
+            $this->core->addSuccessMessage("Requested thread is merged into current thread.");
```

Comment:
This message feels awkward.  Rewrite?  The error was present in the old code, so feel free to ignore this comment if you want.